app/uk/gov/hmrc/gform/sharedmodel/formtemplate/ComponentType.scala

```diff
@@ -97,14 +97,48 @@ case class Address(international: Boolean) extends ComponentType {
 
 }
 
-case object Address {
+object Address {
   val mandatoryFields: FormComponentId => List[FormComponentId] = id => List("street1").map(id.withSuffix)
   val optionalFields: FormComponentId => List[FormComponentId] = id =>
     List("street2", "street3", "street4", "uk", "postcode", "country").map(id.withSuffix)
   val fields: FormComponentId => NonEmptyList[FormComponentId] = id =>
     NonEmptyList.fromListUnsafe(mandatoryFields(id) ++ optionalFields(id))
 }
 
+case class OverseasAddress(
+  mandatoryFields: List[OverseasAddress.Mandatory],
+  value: Option[OverseasAddress.Value]
+) extends ComponentType
+
+object OverseasAddress {
+
+  sealed trait Mandatory
+
+  object Mandatory {
+    case object Line2 extends Mandatory
+    case object City extends Mandatory
+    case object Postcode extends Mandatory
+
+    implicit val format: OFormat[Mandatory] = derived.oformat()
+  }
+
+  case class Value(
+    line1: SmartString,
+    line2: SmartString,
+    line3: SmartString,
+    city: SmartString,
+    postcode: SmartString,
+    country: SmartString
```

Comment:
Why not make this a countrycode field, with limited set of possible country codes?

Reply:
This is supposed to be a free text field. There are no special restriction for `country` according to the ticket.